contracts/simple-ica-controller/src/ibc.rs

```diff
@@ -119,21 +125,20 @@ fn acknowledge_dispatch(
     _deps: DepsMut,
     _env: Env,
     _caller: String,
-    callback: Option<String>,
+    sender: String,
+    callback_id: Option<String>,
     msg: IbcPacketAckMsg,
 ) -> Result<IbcBasicResponse, ContractError> {
-    // TODO: actually handle success/error?
-    match callback {
-        Some(callback) => {
+    match callback_id {
+        Some(id) => {
             // Send IBC packet ack message to another contract
             let msg = WasmMsg::Execute {
-                contract_addr: callback.clone(),
-                msg: to_binary(&ReceiveIbcResponseMsg { msg })?,
+                contract_addr: sender,
+                msg: to_binary(&ReceiveIbcResponseMsg { id, msg })?,
                 funds: vec![],
             };
             Ok(IbcBasicResponse::new()
                 .add_attribute("action", "acknowledge_dispatch")
-                .add_attribute("callback_address", callback)
```

Comment:
We could add callback_id to the event? Not sure if this is useful. Maybe good for debugging.